File: Coding Sample/Java/Advanced/Communication/Distributed/shopping/Store.java
package shopping;

import java.util.*;

class Store{

	private static Random rdm = new Random();
	private static String[] items = {"apple", "mango", "orange", "peach", "pear"};

	public static int findItem(String item){
		return Arrays.binarySearch(items, item);
	}

	public static double priceOf(int id){
		String item = items[id];
		return 0.01 * (rdm.nextInt(1500) + 1000);
	}

	public static int stockOf(int id){
		String item = items[id];
		return 100 + rdm.nextInt(401);
	}
}

File: Coding Sample/MS.NET/Advanced/Distributed/WcfTest/ServerApp/store.cs
using System;

namespace ServerApp
{
	static class Store
	{
		private static Random rdm = new Random();

		internal static int Find(string item)
		{
			string[] items = {"apple", "mango", "orange", "peach", "pear"};

			return Array.IndexOf(items, item.ToLower());
		}

		internal static double PriceOf(int id)
		{
			double[] prices = {21.75, 24.50, 9.25, 5.75, 6.50};

			return prices[id];
		}

		internal static int StockOf(int id)
		{
			return rdm.Next(100, 501);
		}
	}
}
